### DISCUSSION TITLE
STT Request - Azure

### DISCUSSION THREAD

**Discussion by timonvanhasselt** [OpenVoiceOS/ovos-buildroot#159] [Dev]
Would it be possible to write a STT plugin for Azure STT for OVOS? In Mycroft core there was a STT option for Bing, but it is deprecated. 

```
class BingSTT(TokenSTT):
    def __init__(self):
        super(BingSTT, self).__init__()

    def execute(self, audio, language=None):
        self.lang = language or self.lang
        return self.recognizer.recognize_bing(audio, self.token,
                                              self.lang)
```
[(source)](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/stt/__init__.py)

The [TTS plugin ](https://github.com/dalgwen/mycroft-tts-plugin-azure)works great for Azure TTS in Mycroft classic (edit: and now also on OVOS) on my Mark II. 
Maybe the following repository is helpful? An implementation of Azure STT in Python. Unfortunately I can't program in Python myself.
https://github.com/flumi3/speech-to-text

**Comment by JarbasAl** [comment]
done!

https://github.com/OpenVoiceOS/ovos-stt-azure-plugin
https://github.com/OpenVoiceOS/ovos-tts-plugin-azure
